android/walletlib/src/main/java/com/solana/mobilewalletadapter/walletlib/scenario/LocalWebSocketServerScenario.java

```diff
@@ -66,9 +87,19 @@ public void onStarted() {
         public void onFatalError() {
             mIoHandler.post(mCallbacks::onScenarioError);
         }
+
+        @Override
+        public void onNoConnectionTimeoutReached() {
+            if (mCallbacks instanceof Callbacks)
+                ((Callbacks) mCallbacks).onLowPowerAndNoConnectionTimeoutReached();
+        }
     };
 
     private enum State {
         NOT_STARTED, RUNNING, CLOSED
     }
+
+    public interface Callbacks extends Scenario.Callbacks {
+        void onLowPowerAndNoConnectionTimeoutReached();
```

Comment:
NIT: the `TimeoutReached` part of this is an implementation detail. Personally, I'd name this `onLowPowerNoConnection`, and leave how the "no connection" state was determined as an implementation detail.